docs: add tailwindcss for styles, oidc authorize endpoint playground

peintnermax opened the pull request:
Construct `authorize` endpoint to try out different configurations.

mffap commented:
@peintnermax some small observations on first use: 
- I think the button href is not encoded? When I copy the link it works, but seems to have an issue when opening with button (no response type... I guess it escapes after the redirect_uri
- Instance domain trailing slash: we should catch it or describe that we need the trailing slash (easily frustrating)
- the ? after the path should be white

Everything else so far is top!

peintnermax commented:
> @peintnermax some small observations on first use:
> 
> * I think the button href is not encoded? When I copy the link it works, but seems to have an issue when opening with button (no response type... I guess it escapes after the redirect_uri)
> * Instance domain trailing slash: we should catch it or describe that we need the trailing slash (easily frustrating)
> * the ? after the path should be white
> 
> Everything else so far is top!

had a typo.. reponse_type 😂

mffap commented on docs/src/components/authrequest.jsx:
```diff
@@ -0,0 +1,453 @@
+            setInstance(value);
+          }}
+        />
+        <span className={hintClasses}>The domain of your zitadel instance</span>
```
```suggestion
        <span className={hintClasses}>The domain of your zitadel instance with trailing slash.</span>
```

mffap commented on docs/src/components/authrequest.jsx:
```diff
@@ -0,0 +1,453 @@
+                </span>
+              </Listbox.Button>
+              <span className={`${hintClasses} flex`}>
+                This is the resource id of an application. It's the application
+                where you want your users to login.
```
```suggestion
                Determines whether a code, id_token token or just id_token will be returned. Most use cases will need code.
```

mffap commented on docs/src/components/authrequest.jsx:
```diff
@@ -0,0 +1,453 @@
+                  </span>
+                </Listbox.Button>
+                <span className={`${hintClasses} flex`}>
+                  Define if and how the user should be prompted on login.
```
```suggestion
                  Define how the user should be prompted on login and register.
```

mffap commented on docs/src/components/authrequest.jsx:
```diff
@@ -0,0 +1,453 @@
+            }}
+          />
+          <span className={hintClasses}>
+            When requesting the <span className="text-purple-500">scope</span>{" "}
+            <code>urn:zitadel:iam:org:id:{organizationId}`</code>, ZITADEL will
+            enforce that the user is a member of the selected organization
```
```suggestion
            Enforce organization policies and user membership by requesting the <span className="text-purple-500">scope</span>{" "}
            <code>urn:zitadel:iam:org:id:{organizationId}</code>
```

mffap commented on docs/src/components/authrequest.jsx:
```diff
@@ -0,0 +1,453 @@
+      <div className="py-4">
+        <p className="text-sm mt-0 mb-0 text-purple-500">Scopes</p>
+        <span className={`${hintClasses} flex mb-2`}>
+          Request additional information about the user with scopes. The claims
+          (results of scopes) will be returned on the userinfo_endpoint or in
+          the token (when configured).
```
```suggestion
          Request additional information about the user with scopes. The claims will be returned on the userinfo_endpoint or in the token (when configured).
```

mffap commented on docs/src/components/authrequest.jsx:
```diff
@@ -0,0 +1,453 @@
+
+      <br />
+
+      <div className="rounded-md bg-gray-700 shadow dark:bg-black/10 p-2 flex flex-col items-center">
```
Should have a header "Your Authorization Request"

mffap commented on docs/src/components/authrequest.jsx:
```diff
@@ -0,0 +1,453 @@
+      <div className="rounded-md bg-gray-700 shadow dark:bg-black/10 p-2 flex flex-col items-center">
+        <code className="text-sm w-full mb-4 bg-transparent border-none">
+          <span className="text-yellow-500">{instance}</span>
+          <span className="text-white">oauth/v2/authorize</span>
+          <CodeSnipped cname="text-green-500">{`?client_id=${encodeURIComponent(
```
```suggestion
          <span className="text-white">oauth/v2/authorize?</span>
          <CodeSnipped cname="text-green-500">{`client_id=${encodeURIComponent(
```

mffap commented on docs/src/components/authrequest.jsx:
```diff
@@ -0,0 +1,453 @@
+          )}
+        </code>
+
+        <a
+          target="_blank"
+          className="mt-2 flex flex-row items-center py-2 px-4 text-white bg-green-500 dark:bg-green-600 hover:dark:bg-green-500 hover:text-white rounded-md hover:no-underline font-semibold text-sm"
+          href={`${instance}oauth/v2/authorize?client_id=${encodeURIComponent(
+            clientId
+          )}&redirect_uri=${encodeURIComponent(
+            redirectUri
+          )}&response_type=${encodeURIComponent(
+            responseType
+          )}&scope=${encodeURIComponent(scope)}${
+            prompt ? `&prompt=${encodeURIComponent(prompt)}` : ""
+          }${
+            loginHint && prompt === "select_account"
+              ? `&login_hint=${encodeURIComponent(loginHint)}`
+              : ""
+          }`}
+        >
```
Add plausible

peintnermax replied:
done but has to be tested as we have no other window.plausible(...) implemented

mffap commented:
@peintnermax 

- I've added authentication methods with a code challenge
- Moved the playground to its own page
- Added explainer texts

Could you have a quick check, if that looks ok for you? 

The encoding stuff was a bit tricky:`cboa` is apparently deprecated, but I was not able to use the "new" solution, since it required some kind of plugin where webpack was not happy. I think for the moment it's not especially important. I've added the "new" solution as comments.

mffap commented:
@fforootd @livio-a could you have a quick look at the explainer, please?
I didn't want to duplicate the contents of the API docs. The texts should be to give someone without much knowledge an easier introduction to the different parameters.

peintnermax commented:
> @peintnermax
> 
> * I've added authentication methods with a code challenge
> * Moved the playground to its own page
> * Added explainer texts
> 
> Could you have a quick check, if that looks ok for you?
> 
> The encoding stuff was a bit tricky:`cboa` is apparently deprecated, but I was not able to use the "new" solution, since it required some kind of plugin where webpack was not happy. I think for the moment it's not especially important. I've added the "new" solution as comments.



> @peintnermax
> 
> * I've added authentication methods with a code challenge
> * Moved the playground to its own page
> * Added explainer texts
> 
> Could you have a quick check, if that looks ok for you?
> 
> The encoding stuff was a bit tricky:`cboa` is apparently deprecated, but I was not able to use the "new" solution, since it required some kind of plugin where webpack was not happy. I think for the moment it's not especially important. I've added the "new" solution as comments.

I was able to replace the deprecated method by importing from 'buffer'. The rest is done nicely 😎

fforootd commented on docs/docs/apis/openidoauth/authrequest.mdx:
```diff
@@ -0,0 +1,115 @@
+
+The OIDC Playground is for testing OpenID Authentication Requests, giving you more insight how OpenID Connect works and how you can customize ZITADEL behavior with different parameters.
+
+An Open ID Connect (OIDC) [authentication request](https://openid.net/specs/openid-connect-core-1_0.html) is an OAuth 2.0 Authorization Request using additional parameters and scopes to request that the end-user be authenticated by ZITADEL.
```
```suggestion
An OpenID Connect (OIDC) [authentication request](https://openid.net/specs/openid-connect-core-1_0.html) is an OAuth 2.0 Authorization Request using additional parameters and scopes to request that the end-user be authenticated by ZITADEL.
```

fforootd commented on docs/docs/apis/openidoauth/authrequest.mdx:
```diff
@@ -0,0 +1,115 @@
+
+## Why this OIDC playground?
+
+In this version of ZITADEL, human end-users need to authenticate trough a hosted login page.
+Your application should initiate a login by issuing an authentication request and redirecting the user to the login page. You can also customize the behavior of ZITADEL by providing additional parameters and scopes in the request.
```
```suggestion
Currently ZITADEL requires human users to authenticate trough the hosted login page.
Your application should initiate a login by issuing an authentication request and redirecting the user to the login page. You can customize the behavior of ZITADEL by providing additional parameters and scopes in the request.
```

fforootd commented on docs/docs/apis/openidoauth/authrequest.mdx:
```diff
@@ -0,0 +1,115 @@
+In this version of ZITADEL, human end-users need to authenticate trough a hosted login page.
+Your application should initiate a login by issuing an authentication request and redirecting the user to the login page. You can also customize the behavior of ZITADEL by providing additional parameters and scopes in the request.
+
+This playground should help you to initialially craft an authentication request to explore the behavior of ZITADEL in more detail.
```
```suggestion
This playground should help you to initially craft an authentication request and to explore the behavior of ZITADEL in more depth.
```

fforootd commented on docs/docs/apis/openidoauth/authrequest.mdx:
```diff
@@ -0,0 +1,115 @@
+</p>
+
+<p>
+  <span className="text-orange-500">Response Type</span> termines whether a
+  code, id_token token or just id_token will be returned. Most use cases will
+  need code.
```
```suggestion
  <span className="text-orange-500">Response Type</span> defines whether a
  code, id_token token or just id_token will be returned. Most use cases will
  need code.
```

fforootd commented on docs/docs/apis/openidoauth/authrequest.mdx:
```diff
@@ -0,0 +1,115 @@
+
+### Access core apis
+
+Calling the [core API](/docs/apis/introduction) with the authenticated user, requires that the project ID is included in the audience claim.
```
```suggestion
Calling the [core API](/docs/apis/introduction) with the authenticated user, requires that the projectID of ZITADEL is included in the audience claim.
```

fforootd commented on docs/docs/apis/openidoauth/authrequest.mdx:
```diff
@@ -0,0 +1,115 @@
+
+This can be achieved by adding the scope `urn:zitadel:iam:org:project:id:zitadel:aud` to your applications authorization request.
+
+## How to use in your project
```
```suggestion
## How to use ZITADEL in your project
```

fforootd reviewed (commented):
Some small text improvements and corrections, other than that it looks nice

mffap commented:
@peintnermax nothing more from my side for the moment. Anything missing from your end? Can you resolve the conflicts? Thanks

peintnermax commented:
conflicts resolved.. merging is fine from my side 👍

mffap reviewed (approved):
Works like a charm. I think this is good for a first version, we can always improve it later on. Thanks!
